docs/Design/List.md

```diff
@@ -0,0 +1,117 @@
+# List
+
+## Feature Overview
+
+All supported languages provide some amount of built-in list support, either explicitly with lists or with an array equivalent.
+
+
+## Commands
+
+### `list pop`
+
+`list pop : name`
+
+The `list pop` command removes the last element of a list.
+
+## Usage
+
+```gls
+list pop : dogs
+```
+
+### CSharp
+
+```csharp
+dogs.RemoveAt(dogs.Count)
+```
+
+### Java
+
+```java
+dogs.remove(dogs.size()-1)
+```
+
+### Python
+
+```python
+dogs.pop()
+```
+
+### Ruby
+
+```ruby
+dogs.pop
+```
+
+### TypeScript
+
+```typescript
+dogs.pop()
+```
+
+## Implementation
+
+### Properties
+
+These commands will be implemented as native calls.
+
+<table>
+    <thead>
+        <th>Name</th>
+        <th>Type</th>
+        <th>Description</th>
+    </thead>
+    <tbody>
+        <tr>
+            <td>""</td>

```

Comment:
I don't think we need the first property. It's not yet documented but there are already naming properties for arrays/lists.